packages/@webex/plugin-meetings/src/multistream/remoteMediaManager.ts

```diff
@@ -533,9 +575,16 @@ export class RemoteMediaManager extends EventsScope {
     for (let i = 0; i < this.config.audio.numOfActiveSpeakerStreams; i += 1) {
       // eslint-disable-next-line no-await-in-loop
       const slot = await this.receiveSlotManager.allocateSlot(MediaType.AudioMain);
-
       this.slots.audio.push(slot);
     }
+    // create slot for interpretation language audio
+    if (this.config.namedMediaGroup?.value) {
+      const siSlot = await this.receiveSlotManager.allocateSlot(MediaType.AudioMain);
+      if (siSlot) {
+        siSlot.setNamedMediaGroup(this.config.namedMediaGroup);
+        this.slots.audio.push(siSlot);
```

Comment:
I think I mentioned it in the past, but maybe it got lost among the comments: I feel it would be cleaner to store this audio si slot separately instead of appending it to the main audio, so like this:
`this.slots.siAudio = siSlot;`

and actually this makes me realize that then we would have to create a separate RemoteMediaGroup just for the si audio, but it means that we wouldn't need most of the changes you did in RemoteMediaGroup and ReceiveSlot. Basically. we would just add namedMediaGroup as a parameter to RemoteMediaGroup constructor and would construct that simultaneous interpretation remote media group here. We would also need to emit a new type of event to indicate that a new media group for SI is created. What do you think, would that be a better approach?

There is actually a problem with the current implementation that when the app calls setReceiveNamedMediaGroup(), the existing RemoteMediaGroup for main audio gets destroyed and app doesn't know about it and it only gets an event `Event.AudioCreated` out of nowhere with the new group that includes SI audio. If we do the new approach I'm proposing above, we wouldn't need to destroy the main audio group and would only need to emit some new event like "si audio updated" or "si audio destroyed" and "si audio created".

Reply:
According to the design, client need send a complete request (include main and named media if have) out when changing media request. That's why we have to cancel previous request and create a new one when add/change named media group request. We can create a separate RemoteMediaGroup anyway and always keep the main audio RemoteMediaGroup. But will it make logic simpler to destory old one and create a all new  RemoteMediaGroup, instead of maintaining two RemoteMediaGroup  or an array of RemoteMediaGroups in remoteMediaManager?
And it's fine to add a si audio slot IMO.